src/QueryPipeline/QueryPipelineBuilder.h

```diff
@@ -105,6 +115,14 @@ class QueryPipelineBuilder
         bool keep_left_read_in_order,
         Processors * collected_processors = nullptr);
 
+    static std::unique_ptr<QueryPipelineBuilder> joinPipelines2(
```

Comment:
We need a better name here :)

Reply:
Changed to `QueryPipelineBuilder::joinPipelinesYShaped` and `QueryPipelineBuilder::joinPipelinesRightLeft`